### ISSUE TITLE
[Ionic V4.0.0-beta.12] VirtualScroll itemHeight bug

### ISSUE THREAD

**Issue by AIxiuxiu** [ionic-team/ionic#15948]
# Bug Report

**Ionic Info**
Run `ionic info` from a terminal/cmd prompt and paste the output below.

```
Ionic:

   ionic (Ionic CLI)          : 4.1.2 (/Users/xayq-fanxl/.nvm/versions/node/v8.11.2/lib/node_modules/ionic)
   Ionic Framework            : @ionic/angular 4.0.0-beta.12
   @angular-devkit/core       : 0.7.4
   @angular-devkit/schematics : 0.7.4
   @angular/cli               : 6.1.4
   @ionic/ng-toolkit          : 1.0.7
   @ionic/schematics-angular  : 1.0.5

Cordova:

   cordova (Cordova CLI) : 8.0.0
   Cordova Platforms     : android 7.0.0, ios 4.5.5
   Cordova Plugins       : cordova-plugin-ionic-webview 2.1.3, (and 17 other plugins)

System:

   Android SDK Tools : 26.1.1 (/Users/xayq-fanxl/Library/Android/sdk/)
   ios-deploy        : 1.9.2
   ios-sim           : 7.0.0
   NodeJS            : v8.11.2 (/Users/xayq-fanxl/.nvm/versions/node/v8.11.2/bin/node)
   npm               : 6.1.0
   OS                : macOS High Sierra
   Xcode             : Xcode 10.0 Build version 10A255
```

**Describe the Bug**
When I use VirtualScroll, it can't render the header using the `itemHeight` directive.
![image](https://raw.githubusercontent.com/AIxiuxiu/readme_add_pic/master/VirtualScroll2.png)

if i delete the `itemHeight` and then first open the page,it was a blank page and can not render in the most time 
![image](https://raw.githubusercontent.com/AIxiuxiu/readme_add_pic/master/VirtualScroll3.png)
After scroll the page, it will be rendered what i want
![image](https://raw.githubusercontent.com/AIxiuxiu/readme_add_pic/master/VirtualScroll1.png)

while scroll the list，there will always be problems with unrendering or overlapping rendering
![image](https://raw.githubusercontent.com/AIxiuxiu/readme_add_pic/master/VirtualScroll4.png)

**Related Code**
```
<ion-virtual-scroll #virtualScroll [items]="sortedItems" [headerFn]="myHeaderFn" approxHeaderHeight="30px" approxItemHeight="45px">
    <ion-item-divider *virtualHeader="let header" id="scroll-letter-{{ header }}">
        {{ header }}
    </ion-item-divider>
        <ion-item *virtualItem="let item">
            <ion-label color="dark" >
              <ion-text class="row-title">
                  {{item?.name}}
              </ion-text>
            </ion-label>
          </ion-item>
</ion-virtual-scroll>
```
```
myHeaderFn(item, index, items) {
        if (item.isDivider) {
            return item.letter === 'other' ? '#' : item.letter;
        } else {
            return null;
        }
    }

    myItemHeight(item: any, index: number) {
        return 45;
    }
```

**Comment by Lx** [comment]
I'm also experiencing intermittent overlapping of items, especially after navigating backwards:

![image of overlapped ion-virtual-scroll items](https://user-images.githubusercontent.com/160164/47997811-3c981300-e151-11e8-940b-b71707e8fe26.png)

**Comment by Lx** [comment]
Here's an example showing two separate `ion-virtual-scroll` containers, where one is simply zero-height despite containing 20 items, and another superimposes all 21 items in the space of one item:

![image](https://user-images.githubusercontent.com/160164/47998201-96e5a380-e152-11e8-8a28-78ee5d382b67.png)

**Comment by brandyscarney** [comment]
Thanks for the issue! This should be fixed in the `ionic 4.2.0-dev.201903262032.a8a48a4` release. Could you please try this and let us know if it isn't fixed?

**Comment by lucianojdg** [comment]
> Here's an example showing two separate `ion-virtual-scroll` containers, where one is simply zero-height despite containing 20 items, and another superimposes all 21 items in the space of one item:
> 
> ![image](https://user-images.githubusercontent.com/160164/47998201-96e5a380-e152-11e8-8a28-78ee5d382b67.png)


I had the same problem. I can solved by setting "approxItemHeight" property
